cmd/litcli/autopilot.go

```diff
@@ -216,11 +228,27 @@ func initAutopilotSession(ctx *cli.Context) error {
 		}
 	}
 
+	configs := ctx.StringSlice("configuration")
+	features := ctx.StringSlice("feature")
+
+	if len(configs) > 0 && len(features) != len(configs) {
+		return fmt.Errorf("number of features and configurations " +
+			"must match")
+	}
+
 	featureMap := make(map[string]*litrpc.FeatureConfig)
-	for _, feature := range ctx.StringSlice("feature") {
+	for i, feature := range ctx.StringSlice("feature") {
+		var config []byte
+
+		// We allow empty configs, to signal the usage of the default
+		// configuration when the session is registered.
+		if len(configs) > 0 && configs[i] != "{}" {
+			config = []byte(configs[i])
+		}
```

Comment:
nit: Maybe we should also validate here that this is configs[i] is a valid json? After you update this PR to adhere to Elle's firewall storing comment https://github.com/lightninglabs/lightning-terminal/pull/629#discussion_r1300196618, we'll fail there before sending the config to the `AutoPilot server` side, but we'll still execute quite a bit of code and call `ListFeatures` for the `Autopilot server` before getting there. So it might be good to fail early here for something that'll definitely fail.

This isn't blocking either though, so feel free to ignore if you don't think it's worth it.

Reply:
done! agree it's better to check things rather early